lib/AST/Type.cpp

```diff
@@ -134,7 +134,8 @@ bool TypeBase::isUninhabited() {
   // Empty enum declarations are uninhabited
   if (auto nominalDecl = getAnyNominal())
     if (auto enumDecl = dyn_cast<EnumDecl>(nominalDecl))
-      if (enumDecl->getAllElements().empty())
+      // Objective-c enums may yet be inhabited.
```

Comment:
Nitpick: "Objective-C", but really I think this needs more of an explanation, like "in practice, C enums can have any value that fits in their underlying type".